src/test/java/org/cactoos/scalar/AndInThreadsTest.java

```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testProcIterable() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        new AndInThreads(
+            new Proc.NoNulls<Integer>(list::add),
+            Arrays.asList(1, 2)
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testIteratorScalarBoolean() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                new ListOf<Scalar<Boolean>>(
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(true)
+                    ),
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(false)
+                    )
+                ).iterator()
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    @Test
+    public void testIterableScalarBoolean() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                new ListOf<Scalar<Boolean>>(
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(true)
+                    ),
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(false)
+                    )
+                )
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    @Test
+    public void testExecServiceProcValues() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        final ExecutorService service = Executors.newSingleThreadExecutor();
+        new AndInThreads(
+            service,
+            new Proc.NoNulls<Integer>(list::add),
+            1, 2
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testExecServiceProcIterator() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        new AndInThreads(
+            Executors.newSingleThreadExecutor(),
+            new Proc.NoNulls<Integer>(list::add),
+            Arrays.asList(1, 2).iterator()
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testExecServiceProcIterable() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        final ExecutorService service = Executors.newSingleThreadExecutor();
+        new AndInThreads(
```

Comment:
@driver733 What is the returned ``value()`` here and why?

Reply:
@amihaiemil The result is always true. This has to be documented in the source class.
#688